# Análisis Exploratorio de Data Product 6

Se tiene registro [(Data Product 1)](../Data/DP1_covid19-peru_x_semanaEpi.csv) de que las primeras dosis recibidas fueron a partir de la SE37 del 2020, posiblemente grupos vulnerables o personas que fueron vacunadas en el extranjero. Debido a que la **vacunación masiva oficialmente comenzó la SE6 del 2021**. De igual manera segundas y terceras dosis en SE9 y SE41 del 2021 respectivamente. *(Hasta aquí todo correcto).*

Sin embargo el problema detectado en [Data Product 6 (DP6)](../Data/DP6_covid19-peru_info-fallecidos.csv) de casos individuales de fallecidos, deja notar que existen registros de personas que recibieron la 1era y 2nda dosis luego de la fecha de fallecimiento. 

**Posibles escenarios:**
- Registros erróneos de fallecimientos *(El más posible)*
- Registros erróneos del id de cada persona
- Registros erróneos en la fecha de vacunación

**Analizando los resultados de DP6:**

In [1]:
import pandas as pd
dp6 = pd.read_csv('../Data/DP6_covid19-peru_info-fallecidos.csv')

# Convertimos en formato fecha las columnas de vacunación
lst_dates = ['fecha_vacuna_1', 'fecha_vacuna_2', 'fecha_vacuna_3']
for date in lst_dates:
    dp6.loc[:,date] = pd.to_datetime(dp6[date], format = '%Y%m%d')
    
dp6 

id_persona fecha_fallecimiento  year  epi_week  time_days  sexo  edad  \
0       36527082.0          2020-03-03  2020        10          1     1    63   
1       36401995.0          2020-03-04  2020        10          2     1    85   
2       36853977.0          2020-03-07  2020        10          5     2    24   
3              NaN          2020-03-11  2020        11          9     2    78   
4       36614476.0          2020-03-14  2020        11         12     1     4   
...            ...                 ...   ...       ...        ...   ...   ...   
203371   9346434.0          2022-01-14  2022         2        683     2    59   
203372  11184318.0          2022-01-14  2022         2        683     1    64   
203373   8196728.0          2022-01-14  2022         2        683     1    64   
203374  17398883.0          2022-01-14  2022         2        683     1    64   
203375  32879545.0          2022-01-14  2022         2        683     2    69   

        edad_cat  departamento  vacunado  dosis_1 fecha_vacuna_1  id_vacuna_1  \
0              5            22         0        0            NaT            0   
1              7            22         0        0            NaT            0   
2              1            16         0        0            NaT            0   
3              6            20         0        0            NaT            0   
4              0            15         0        0            NaT            0   
...          ...           ...       ...      ...            ...          ...   
203371         4            15         0        0            NaT            0   
203372         5            15         1        1     2021-02-25            6   
203373         5            15         1        1     2021-06-01            5   
203374         5             8         1        1     2021-06-02            5   
203375         5            17         1        1     2021-06-12            2   

        dosis_2 fecha_vacuna_2  id_vacuna_2  dosis_3 fecha_vacuna_3  \
0             0            NaT            0        0            NaT   
1             0            NaT            0        0            NaT   
2             0            NaT            0        0            NaT   
3             0            NaT            0        0            NaT   
4             0            NaT            0        0            NaT   
...         ...            ...          ...      ...            ...   
203371        0            NaT            0        0            NaT   
203372        1     2021-03-18            6        0            NaT   
203373        1     2021-06-23            5        1     2021-12-13   
203374        1     2021-06-25            5        0            NaT   
203375        1     2021-10-06            3        0            NaT   

        id_vacuna_3  
0                 0  
1                 0  
2                 0  
3                 0  
4                 0  
...             ...  
203371            0  
203372            0  
203373            5  
203374            0  
203375            0  

[203376 rows x 19 columns]

Filtramos todos los casos donde la primera fecha de vacunación sea mayor a la fecha de fallecimiento.

In [2]:
filtered_data = dp6[(dp6['fecha_vacuna_1'] >= dp6['fecha_fallecimiento'])]
filtered_data

id_persona fecha_fallecimiento  year  epi_week  time_days  sexo  edad  \
6       24907449.0          2020-03-16  2020        12         14     2    26   
235     18821159.0          2020-04-03  2020        14         32     1    34   
251      3325727.0          2020-04-03  2020        14         32     1    69   
260      6570051.0          2020-04-03  2020        14         32     2    37   
289      4157809.0          2020-04-04  2020        14         33     1    45   
...            ...                 ...   ...       ...        ...   ...   ...   
198290  14764120.0          2021-08-24  2021        34        540     2    16   
198570  30546838.0          2021-08-30  2021        35        546     1    32   
198875  22737644.0          2021-09-07  2021        36        554     2    76   
198892  24691491.0          2021-09-08  2021        36        555     2    37   
200764    719712.0          2021-11-07  2021        45        615     2    42   

        edad_cat  departamento  vacunado  dosis_1 fecha_vacuna_1  id_vacuna_1  \
6              1             5         1        1     2021-10-10            6   
235            2             7         1        1     2021-08-28            6   
251            5            15         1        1     2021-12-04            5   
260            2            15         1        1     2021-02-11            6   
289            3            15         1        1     2021-07-13            5   
...          ...           ...       ...      ...            ...          ...   
198290         0            20         1        1     2021-10-12            3   
198570         2            20         1        1     2021-10-01            3   
198875         6             8         1        1     2021-09-25            6   
198892         2            20         0        1     2021-09-19            5   
200764         3            24         0        1     2022-01-02            6   

        dosis_2 fecha_vacuna_2  id_vacuna_2  dosis_3 fecha_vacuna_3  \
6             1     2021-10-31            6        0            NaT   
235           1     2021-09-19            6        0            NaT   
251           1     2021-12-26            5        0            NaT   
260           1     2021-03-04            6        1     2021-11-03   
289           1     2021-08-04            5        0            NaT   
...         ...            ...          ...      ...            ...   
198290        1     2021-11-09            1        0            NaT   
198570        1     2021-12-19            1        0            NaT   
198875        1     2021-11-25            6        0            NaT   
198892        0            NaT            0        0            NaT   
200764        0            NaT            0        0            NaT   

        id_vacuna_3  
6                 0  
235               0  
251               0  
260               5  
289               0  
...             ...  
198290            0  
198570            0  
198875            0  
198892            0  
200764            0  

[1646 rows x 19 columns]

In [3]:
print(f'Reportes donde la fecha de vacunación es posterior a la fecha de fallecimiento {len(filtered_data)}')
print(f'Representa a {round(len(filtered_data)*100/len(dp6),2)}% del total de fallecidos')

Reportes donde la fecha de vacunación es posterior a la fecha de fallecimiento 1646
Representa a 0.81% del total de fallecidos


- Existen un total de **1646 casos** donde la fecha de vacunación es posterior a la fecha de fallecimiento
- Es decir un **0.81%** del total de fallecidos. 
- **Menos del 1%**

Tomamos 10 casos aleatorios para evaluar

In [28]:
sample = filtered_data.sample(10, random_state = 1).sort_values(by = 'id_persona')
sample = sample[['id_persona', 'fecha_fallecimiento', 'edad', 'dosis_1', 'dosis_2', 'dosis_2',
                 'fecha_vacuna_1', 'fecha_vacuna_2', 'fecha_vacuna_3']]
sample

id_persona fecha_fallecimiento  edad  dosis_1  dosis_2  dosis_2  \
169333   1423229.0          2021-05-02    76        1        0        0   
91383    2295115.0          2020-12-12    91        1        1        1   
133875   2416721.0          2021-03-16    51        1        1        1   
137221   4298542.0          2021-03-21    61        1        1        1   
8699    13354393.0          2020-05-08    41        1        1        1   
51028   16716381.0          2020-07-22    75        1        1        1   
48126   22603584.0          2020-07-17    61        1        1        1   
91133   23967370.0          2020-12-10    78        1        1        1   
185742  29692043.0          2021-06-01    61        1        0        0   
94970   36223908.0          2021-01-06    77        1        0        0   

       fecha_vacuna_1 fecha_vacuna_2 fecha_vacuna_3  
169333     2021-08-03            NaT            NaT  
91383      2021-04-23     2021-05-14     2021-11-30  
133875     2021-07-04     2021-08-02     2022-01-07  
137221     2021-06-10     2021-07-03            NaT  
8699       2021-08-10     2021-09-02     2022-01-06  
51028      2021-05-14     2021-06-08     2021-12-04  
48126      2021-06-14     2021-07-05            NaT  
91133      2021-05-03     2021-05-25     2021-11-26  
185742     2021-06-09            NaT            NaT  
94970      2021-08-12            NaT            NaT

Como se logra apreciar varios de los casos reportados como "fallecidos" tienen hasta 3 dosis recibidas. Por lo que posiblemente haya sido un error registrarlos como fallecimientos.

Sin embargo estos 10 casos aleatorios se los evaluará individualmente para verificar que se traten de la misma persona.

La información personal (especialmente importante el id) se la extrajo del registro de [Fallecidos por COVID-19](https://www.datosabiertos.gob.pe/dataset/fallecidos-por-covid-19-ministerio-de-salud-minsa). Por lo que se procede a comparar con los registros de [Vacunación contra la COVID-19](https://www.datosabiertos.gob.pe/dataset/vacunacion) de acuerdo a su id para verificar que sean la misma persona.

**Se extraen estos 10 id de la muestra aleatoria:**

In [23]:
cases = list(sample.id_persona)

**Comparamos el id de cada uno con el id del dataset de vacunación:**

In [30]:
import dask.dataframe as dd
vacunados = dd.read_csv('../RawData/TB_VACUNACION_COVID19.csv', dtype={'edad': 'float64'},
                        usecols = ['id_persona', 'fecha_vacunacion', 'dosis', 'edad'])

vacunados = vacunados[vacunados['id_persona'].isin(cases)].compute().sort_values(by = ['id_persona', 'dosis'])

In [31]:
vacunados

id_persona  fecha_vacunacion  dosis  edad
1080677     1423229          20210803      1  76.0
221400      2295115          20210423      1  91.0
569847      2295115          20210514      2  91.0
304775      2295115          20211130      3  92.0
70512       2416721          20210704      1  52.0
438470      2416721          20210802      2  52.0
426035      2416721          20220107      3  52.0
983536      4298542          20210610      1  62.0
425905      4298542          20210703      2  62.0
277803     13354393          20210810      1  42.0
237297     13354393          20210902      2  42.0
253864     13354393          20220106      3  42.0
1093747    16716381          20210514      1  76.0
206838     16716381          20210608      2  76.0
87731      16716381          20211204      3  77.0
598767     22603584          20210614      1  62.0
454481     22603584          20210705      2  62.0
179478     23967370          20210503      1  79.0
1313279    23967370          20210525      2  79.0
1050022    23967370          20211126      3  79.0
1351598    29692043          20210609      1  61.0
1098113    36223908          20210812      1  77.0

In [29]:
sample

id_persona fecha_fallecimiento  edad  dosis_1  dosis_2  dosis_2  \
169333   1423229.0          2021-05-02    76        1        0        0   
91383    2295115.0          2020-12-12    91        1        1        1   
133875   2416721.0          2021-03-16    51        1        1        1   
137221   4298542.0          2021-03-21    61        1        1        1   
8699    13354393.0          2020-05-08    41        1        1        1   
51028   16716381.0          2020-07-22    75        1        1        1   
48126   22603584.0          2020-07-17    61        1        1        1   
91133   23967370.0          2020-12-10    78        1        1        1   
185742  29692043.0          2021-06-01    61        1        0        0   
94970   36223908.0          2021-01-06    77        1        0        0   

       fecha_vacuna_1 fecha_vacuna_2 fecha_vacuna_3  
169333     2021-08-03            NaT            NaT  
91383      2021-04-23     2021-05-14     2021-11-30  
133875     2021-07-04     2021-08-02     2022-01-07  
137221     2021-06-10     2021-07-03            NaT  
8699       2021-08-10     2021-09-02     2022-01-06  
51028      2021-05-14     2021-06-08     2021-12-04  
48126      2021-06-14     2021-07-05            NaT  
91133      2021-05-03     2021-05-25     2021-11-26  
185742     2021-06-09            NaT            NaT  
94970      2021-08-12            NaT            NaT

In [3]:
import pandas as pd
df = pd.read_csv('../Data/DP3_covid19-peru_casos-positivos_+_info-fallecimiento_+_info-vacunas.csv')

In [5]:
df[df['fallecido'] == 1]

id_persona  fecha_resultado  metododx  sexo  edad  edad_cat  \
8          36889651         20200312         2     2    75         6   
9          36611013         20200312         2     1    78         6   
46         36450848         20200315         2     1    96         7   
62         36539210         20200316         2     1    71         6   
99         36818328         20200318         2     2    63         5   
...             ...              ...       ...   ...   ...       ...   
3245621    32154742         20220308         2     2    58         4   
3246233    21242745         20220309         2     2    76         6   
3246645      967423         20220309         2     1    77         6   
3247171     7457370         20220310         2     1    52         4   
3247246    34567900         20220310         2     2    66         5   

         departamento  fallecido  time_days  fecha_fallecimiento  ...  \
8                15.0          1          7           20200320.0  ...   
9                15.0          1          6           20200319.0  ...   
46               15.0          1         16           20200329.0  ...   
62                7.0          1         32           20200414.0  ...   
99               15.0          1         47           20200429.0  ...   
...               ...        ...        ...                  ...  ...   
3245621          15.0          1        726           20220309.0  ...   
3246233          24.0          1        706           20220217.0  ...   
3246645          21.0          1        720           20220303.0  ...   
3247171          14.0          1        707           20220218.0  ...   
3247246          21.0          1        727           20220310.0  ...   

         vacunado  dosis_1  dosis_2  dosis_3  id_vacuna_1  id_vacuna_2  \
8               0        0        0        0            0            0   
9               0        0        0        0            0            0   
46              0        0        0        0            0            0   
62              0        0        0        0            0            0   
99              0        0        0        0            0            0   
...           ...      ...      ...      ...          ...          ...   
3245621         1        1        1        0            6            6   
3246233         0        0        0        0            0            0   
3246645         1        1        1        1            1            1   
3247171         1        1        1        1            5            5   
3247246         1        1        1        0            6            6   

         id_vacuna_3  fecha_vacuna_1  fecha_vacuna_2  fecha_vacuna_3  
8                  0             NaN             NaN             NaN  
9                  0             NaN             NaN             NaN  
46                 0             NaN             NaN             NaN  
62                 0             NaN             NaN             NaN  
99                 0             NaN             NaN             NaN  
...              ...             ...             ...             ...  
3245621            0      20210915.0      20211006.0             NaN  
3246233            0             NaN             NaN             NaN  
3246645            5      20210603.0      20210701.0      20211230.0  
3247171            5      20210526.0      20210617.0      20220113.0  
3247246            0      20210910.0      20211001.0             NaN  

[101697 rows x 22 columns]

**Verificando la edad de las personas podemos comprobar que todas las id corresponden a la mismas personas, podemos decir que 0.81% de los datos de fallecidos son erróneos**